Source/Core/VideoCommon/Fifo.cpp

```diff
@@ -370,30 +362,14 @@ void RunGpuLoop()
 				AsyncRequests::GetInstance()->PullEvents();
 			}
 
+			// We are likely shutting down, so use our free time to do as much work as possible.
```

Comment:
How about " // The fifo is empty and it's unlikely we will get any more work in the near future. Make sure VertexManager finishes drawing any primitives it has stored in it's buffer."